internal/connector/internal/config/connectors.go

```diff
@@ -79,22 +148,29 @@ func (c *ConnectorsConfig) ReadFiles() error {
 			// Read the file
 			buf, err := os.ReadFile(path)
 			if err != nil {
-				err = gherrors.Errorf("error reading catalog file %s: %s", path, err)
-				return err
+				return gherrors.Errorf("error reading catalog file %s: %s", path, err)
 			}
 
 			entry := ConnectorCatalogEntry{}
 			err = json.Unmarshal(buf, &entry)
 			if err != nil {
-				err = gherrors.Errorf("error unmarshaling catalog file %s: %s", path, err)
-				return err
+				return gherrors.Errorf("error unmarshaling catalog file %s: %s", path, err)
+			}
+
+			// set catalog metadata from metadata config read earlier
+			id := entry.ConnectorType.Id
+			if meta, found := connectorMetadata[id]; found {
+				entry.ConnectorType.FeaturedRank = meta.FeaturedRank
+				entry.ConnectorType.Labels = meta.Labels
+				// TODO annotations
+			} else {
+				return gherrors.Errorf("Missing metadata for connector %s", id)
 			}
 
 			// compute checksum for catalog entry to look for updates
 			sum, err := checksum(entry)
 			if err != nil {
-				err = gherrors.Errorf("error computing checksum for catalog file %s: %s", path, err)
-				return err
+				return gherrors.Errorf("error computing checksum for catalog file %s: %s", path, err)
```

Comment:
You should be using `%w` verb instead of `%s` so that the underlaying error is correctly wrapped.